MGDOBR-985: add error handling cypress tests

jomarko opened the pull request:
Dependent on #145

sjamboro commented on cypress/integration/InstancesTest.ts:
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
+          .should("contain.text", "Error handling method")
+          .should("be.visible");
+
+        cy.get(".pf-c-description-list__term")
```
I was able to add ouiaId to each component. See:
![image](https://user-images.githubusercontent.com/1536609/193225878-78410b1a-a966-4448-b707-d4898a6ac652.png)

There are the desired elements/components:
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/main/src/app/Processor/ProcessorConfigParameters/ProcessorConfigParameters.tsx#L29
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/main/src/app/Processor/ProcessorConfigParameters/ProcessorConfigParameters.tsx#L50

jomarko replied:
I will try, sure, thank you

jomarko commented on cypress/integration/InstancesTest.ts:
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
+        cy.get(".pf-c-description-list__term")
+          .should("have.length.gt", 1)
+          .should(($items) => {
+            expect($items).to.have.length(5);
```
@sjamboro yes, thank you for a question, during the test creation, I have an impression that there is some delay between opening the details panel, and showing all attributes. that is why I constructed the check in this way.

Please tell me what is a desired state form you point of view:
- remove `.should("have.length.gt", 1)` check
- replace `.should("have.length.gt", 1)` check with `.should("have.length", 5)`
- replace `.should("have.length.gt", 1)` check with ...
- ...

?

jomarko commented on cypress/integration/InstancesTest.ts:
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
+          .should("have.length.gt", 1)
+          .should(($items) => {
+            expect($items).to.have.length(5);
+            expect($items.eq(0)).to.contain.text("Error handling method");
```
@sjamboro just personal preference, to make assertion safe from changes like:
- "Error handling method:"
- "Error handling method -"
- "[icon] Error handling method"
- ...

I can use `have`if you wish

sjamboro replied:
@jomarko yes I understand but my personal preferences are opposite :D (I would like to know that the label is changed)

jomarko commented:
@sjamboro please re-review #169. I believe I addressed most your comments. Some of my changes may change your opinion about some point I didn't address. Please let me know what are your thoughts about the PR now.

ChughShilpa commented on cypress/integration/InstancesTest.ts:
```diff
@@ -272,6 +272,82 @@ describe("Instances Test", () => {
+            cy.get("[data-testid='basic_auth_password-value']").should(
+              "contain.text",
+              "****"
+            );
```
IMO, Password must be asserted with 'have' as this value is constant

ChughShilpa replied:
Same opinion for other fields as well.

jomarko replied:
pushing in a separate commit, thank you

sjamboro reviewed (approved):
Thank you ... It looks good.

Please check that all tests are performed in GHA (few minutes ago there was performed only Instance Error Handler test).

ChughShilpa reviewed (approved):
Thanks for the update

sjamboro commented:
> @jomarko LGTM. thanks. Few observation/queries not from the PR Do we have test coverage for instance creation with error handler type and also noticed that for Error Handling in UI we don't have option to edit the parameters but our API supports

I do not think so. Error Handling is quite new in UI.

jomarko commented:
@sjamboro yes, now tests are executed. It had pushed `it.only` by mistake. Fixed by force push.

@Srihari1192 about the edit error handler configuration update. It is opened right now. I added you as reviewer https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/204

@Srihari1192 creation of RHOSE instance with error handling strategy is not covered. Please create a jira and feel free to contribute with such  test. Please be aware of realted test that is pending from me https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/203

Srihari1192 commented:
> @sjamboro yes, now tests are executed. It had pushed `it.only` by mistake. Fixed by force push.
> 
> @Srihari1192 about the edit error handler configuration update. It is opened right now. I added you as reviewer #204
> 
> @Srihari1192 creation of RHOSE instance with error handling strategy is not covered. Please create a jira and feel free to contribute with such test. Please be aware of realted test that is pending from me #203

@jomarko Sure thanks for the info
